keps/sig-network/20190603-EndpointSlice-API.md

```diff
@@ -59,81 +59,118 @@ type EndpointSlice struct {
+    // following well known keys:
+    // * kubernetes.io/hostname: the value indicates the hostname of the node
+    //   where the endpoint is located. This should match the corresponding
+    //   node label.
+    // * topology.kubernetes.io/zone: the value indicates the zone where the
+    //   endpoint is located. This should match the corresponding node label.
+    // * topology.kubernetes.io/region: the value indicates the region where the
+    //   endpoint is located. This should match the corresponding node label.
+    // +optional
     Topology map[string]string `json:"topology,omitempty" protobuf:"bytes,5,opt,name=topology"`
 }
 
+// EndpointConditions represents the current condition of an endpoint.
 type EndpointConditions struct {
-    // Ready indicates if the endpoint is ready to serve traffic.
-    Ready bool `json:"ready,omitempty" protobuf:"bytes,1,opt,name=ready"`
-}
-
-type EndpointMeta struct {
-    // This field specifies the list of ports associated with each endpoint in the EndpointSlice
-    // Each EndpointPort must have a unique port name.
-    // +optional 
-    // +listType=atomic
-    Ports []EndpointPort `json:"ports,omitempty" protobuf:"bytes,1,opt,name=ports"`
+    // ready indicates that this endpoint is prepared to receive traffic,
+    // according to whatever system is managing the endpoint. A nil value
+    // indicates an unknown state. In most cases consumers should interpret this
+    // unknown state as ready.
+    // +optional
+    Ready *bool `json:"ready,omitempty" protobuf:"bytes,1,name=ready"`
 }
 
+// EndpointPort represents a Port used by an EndpointSlice
 type EndpointPort struct {
-    // The name of this port. 
-    // If the EndpointSlice is dervied from K8s service, this corresponds to ServicePort.Name.
-    // Name must be a IANA_SVC_NAME or an empty string.
+    // The name of this port. All ports in an EndpointSlice must have a unique
+    // name. If the EndpointSlice is dervied from a Kubernetes service, this
+    // corresponds to the Service.ports[].name.
+    // Name must either be an empty string or pass IANA_SVC_NAME validation:
+    // * must be no more than 15 characters long
+    // * may contain only [-a-z0-9]
+    // * must contain at least one letter [a-z]
+    // * it must not start or end with a hyphen, nor contain adjacent hyphens
     // Default is empty string.
-    // +optional 
-    Name *string `json:"name,omitempty" protobuf:"bytes,1,opt,name=name"`
+    Name *string `json:"name,omitempty" protobuf:"bytes,1,name=name"`
     // The IP protocol for this port.
     // Must be UDP, TCP, or SCTP.
     // Default is TCP.
-    // +optional 
-    Protocol *v1.Protocol `json:"protocol,omitempty" protobuf:"bytes,2,opt,name=protocol"`
+    Protocol *v1.Protocol `json:"protocol,omitempty" protobuf:"bytes,2,name=protocol"`
+    // The application protocol for this port.
+    // +optional
+    AppProtocol *string `json:"protocol,omitempty" protobuf:"bytes,2,name=protocol"`
```

Comment:
json/proto name should not be `protocol`, probably `appProtocol` instead?

Reply:
Woops, fixed, thanks!